From 872c42010793c2977acf7e6dd436d48028659f3e Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Mon, 22 Nov 2010 14:29:53 +0100
Subject: ASoC: ABE DSP: Restore context for ABE registers after retention

Update driver with ABE wake up function, that will restore context
of ABE registers which may be lost during retention.

Change-Id: I0b60619c03d55d835a59a733c23a343ceaf6636a
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
Signed-off-by: Chris Kelly <c-kelly@ti.com>
---
 sound/soc/omap/omap-abe-dsp.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:14:41.963323364 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-19 00:14:44.212071001 +0530
@@ -1696,8 +1696,8 @@
 
 	pm_runtime_get_sync(&pdev->dev);
 
-	abe_reset_hal();
-	abe_write_event_generator(EVENT_TIMER);
+	if (!abe->active++)
+		abe_wakeup();
 
 	switch (dai->id) {
 	case ABE_FRONTEND_DAI_MODEM:
@@ -1709,8 +1709,6 @@
 		break;
 	}
 
-	abe->active++;
-
 	mutex_unlock(&abe->mutex);
 	return 0;
 }
